database-commons/src/main/java/io/cdap/plugin/db/batch/action/ArgumentSetterConfig.java

```diff
@@ -0,0 +1,154 @@
+  @Description("JDBC connection string including database name.")
+  @Macro
+  public String password;
+
+
+  @Name(DATABASE_NAME)
+  @Description("The name of the database which contains\n"
+      + "the configuration table")
+  @Macro
+  String databaseName;
+
+  @Name(TABLE_NAME)
+  @Description("The name of the table in the database\n"
+      + "containing the configurations for the pipeline")
+  @Macro
+  String tableName;
+
+  @Name(ARGUMENT_SELECTION_CONDITIONS)
+  @Description("A set of conditions for identifying the\n"
+      + "arguments to run a pipeline. Users can\n"
+      + "specify multiple conditions in the format\n"
+      + "column1=<column1-value>;column2=<colum\n"
+      + "n2-value>. A particular use case for this\n"
+      + "would be feed=marketing AND\n"
+      + "date=20200427. The conditions specified\n"
+      + "should be logically ANDed to determine the\n"
+      + "arguments for a run. When the conditions are\n"
+      + "applied, the table should return exactly 1 row.\n"
+      + "If it doesn’t return any rows, or if it returns\n"
+      + "multiple rows, the pipeline should abort with\n"
+      + "appropriate errors. Typically, users should\n"
+      + "use macros in this field, so that they can\n"
+      + "specify the conditions at runtime.")
+  @Macro
+  String argumentSelectionConditions;
+
+  @Name(ARGUMENTS_COLUMNS)
+  @Description("Names of the columns that contain the\n"
+      + "arguments for this run. The values of this\n"
+      + "columns in the row that satisfies the argument\n"
+      + "selection conditions determines the\n"
+      + "arguments for the pipeline run")
+  @Macro
+  String argumentsColumns;
+
+  public String getDatabaseName() {
+    return databaseName;
+  }
+
+  public String getTableName() {
+    return tableName;
+  }
+
+  public String getArgumentSelectionConditions() {
+    return argumentSelectionConditions;
+  }
+
+  public String getArgumentsColumns() {
+    return argumentsColumns;
+  }
+
+  public String getQuery() {
+    if (this.getArgumentSelectionConditions() == null) {
+      throw new IllegalArgumentException("Argument selection conditions is are empty.");
+    }
+    String[] split = this.getArgumentSelectionConditions().split(";");
+    String conditions = String.join(" AND ", split);
+
+    return String
+        .format("SELECT %s FROM %s WHERE %s", this.getArgumentsColumns(), this.getTableName(),
+            conditions);
+  }
+
+  /**
+   * Validates config input fields.
+   *
+   * @param collector context failure collector {@link FailureCollector}
+   */
+  public void validate(FailureCollector collector) {
+    if (!containsMacro(CONNECTION_STRING) && Strings.isNullOrEmpty(this.connectionString)) {
+      collector.addFailure("Invalid connection string", "Connection string cannot be empty");
+    }
+    if (!containsMacro(ConnectionConfig.USER) && Strings.isNullOrEmpty(this.user)) {
+      collector.addFailure("Invalid username", "Username cannot be empty");
```

Comment:
please add period at the end of every failure message and its cause

Reply:
@CuriousVini updated all messages